dash_auth/oauth.py

```diff
@@ -238,11 +240,25 @@ def serve_oauth_redirect(self):
 
     def set_cookie(self, response, name, value, max_age,
                    httponly=True, samesite='Strict'):
+
+        secure = True if 'https:' in self._app_url else False
+
+        is_http = flask.request.environ.get(
+            'wsgi.url_scheme',
+            flask.request.environ.get('HTTP_X_FORWARDED_PROTO', 'http')
+        ) == 'http'
+
+        ua = user_agent_parser.ParseUserAgent(
+            flask.request.environ.get('HTTP_USER_AGENT', ''))
```

Comment:
This should default to `{}` not `''` or else the `ua.get()` call is going to fail in the default case.

Reply:
I'm not sure you understand my concern here.

If `flask.request.environ.get('HTTP_USER_AGENT', ''))` fails to find the `HTTP_USER_AGENT` key, it will return the default value of `''`, which will be assigned to `ua`.

Your next statement does `ua.get('family')`, which will fail in that case:

```
In [1]: ua = ''

In [2]: print(ua.get('family'))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-64d5a5c22c32> in <module>()
----> 1 print(ua.get('family'))

AttributeError: 'str' object has no attribute 'get'
```

If you use `{}` instead, this problem is avoided:

```
In [3]: ua = {}

In [4]: print(ua.get('family'))
None
```

Reply:
`flask.requests.environ.get('HTTTP_USER_AGENT', '') is passed as parameter to `user_agent_parser.ParseUserAgent` on the line before, it takes a string not a dict and return a dict.

Reply:
Ah, right, sorry I missed that.